server/testdb/testdb.go

```diff
@@ -0,0 +1,41 @@
+package testdb
+
+import (
+	"context"
+	"errors"
+
+	"github.com/kubeshop/tracetest/openapi"
+)
+
+var ErrNotFound = errors.New("record not found")
+
+type Tests interface {
```

Comment:
I don't think those `Tests`, `Results`, and `Assertions` names are good. They look like a definition of a model, not a repository. 

Could we add something else in their names to ensure it is explicit their function? Like `TestsRepository`.